app/models/court_application_type.rb

```diff
@@ -28,6 +32,13 @@ class CourtApplicationType < ApplicationRecord
   validates :spi_out_applicable_flag, presence: true
   validates :offence_active_order, presence: true
 
+  before_save do
+    unless code.nil?
+      self.category_code = COURT_APPLICATION_TYPES[code]&.fetch("category", nil)
+      self.type = COURT_APPLICATION_TYPES[code]&.fetch("title", nil)
+    end
+  end
+
```

Comment:
I think it's better to move the assigmements inside the controller, in`CourtApplicationTypesController#update`